hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -1055,6 +1063,69 @@ public void close() throws Exception {
    */
   protected abstract void commit(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap);
 
+  /**
+   * Converts the input records to the input format expected by the write client.
+   * @param records records to be converted
+   * @return converted records
+   */
+  protected abstract I convertRecordsToWriteClientInput(HoodieData<HoodieRecord> records);
+
+  protected void commitInternal(String instantTime, Map<MetadataPartitionType, HoodieData<HoodieRecord>> partitionRecordsMap, boolean isInitializing,
+                                Option<BulkInsertPartitioner> bulkInsertPartitioner) {
```

Comment:
Is partitioner necessary here? Why can't we bulk insert from the engine-specific implemenation?

Reply:
The bulk insert methods take in a partitioner

Reply:
I assume we are overriding this for flink since it has some minor differences. jsut calling it out. in the interest of addressing any feedback, we should not revert that to use this common method.

Reply:
Yes it was originally to try to get flink using common code since the flink code does not have partitioner but was using bulk insert while initializing the table

Reply:
so, did we remove the new argument and re-use the presence of partitioner?

Reply:
Right now I'm still using the new argument so I can just have flink override this method until it can be moved on to the common implementation.